all-spark-notebook/Dockerfile

```diff
@@ -90,12 +90,11 @@ RUN conda install --yes \
 RUN mkdir -p /opt/conda/share/jupyter/kernels/scala
 COPY kernel.json /opt/conda/share/jupyter/kernels/scala/
 
-USER root
-
-# Install Python 2 kernel spec globally to avoid permission problems when NB_UID
-# switching at runtime.
-RUN $CONDA_DIR/envs/python2/bin/python \
-    $CONDA_DIR/envs/python2/bin/ipython \
-    kernelspec install-self
-
-USER jovyan
+# Install Python 2 kernel spec into the Python 3 conda environment which
+# runs the notebook server
+RUN bash -c '. activate python2 && \
+    python -m ipykernel.kernelspec --prefix=$CONDA_DIR && \
+    . deactivate'
+# Set PYSPARK_HOME in the python2 spec
+RUN sed -i '$ d' $CONDA_DIR/share/jupyter/kernels/python2/kernel.json && \
+    echo ', "env": {"PYSPARK_PYTHON": "/opt/conda/envs/python2/bin/python"} }' >> $CONDA_DIR/share/jupyter/kernels/python2/kernel.json
```

Comment:
I agree with you. Good recommendation. The only quirky part is apt-get installing jq in the dockerfile, doing this, then taking jq back out. So in the end it becomes:

```
RUN apt-get -y update && \
    apt-get -y install jq && \
    jq --arg v "$CONDA_DIR/envs/python2/bin/python" \
        '.["env"]["PYSPARK_PYTHON"]=$v' \
        $CONDA_DIR/share/jupyter/kernels/python2/kernel.json
    apt-get remove -y jq && \
    apt-get clean
```

We could leave jq there, but it's a bit odd to only have it in the pyspark and all-spark notebooks. If anything, it belongs in minimal-notebook, but I see that as a step to take separate from this PR.